src/external_printer.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(feature = "external_printer")]
+use crossbeam::channel::{bounded, Receiver, Sender};
+
+/// An ExternalPrinter allows to print messages of text while editing a line.
+/// The message is printed as a new line, the line-edit will continue below the
+/// output.
+#[cfg(feature = "external_printer")]
+#[derive(Debug, Clone)]
+pub struct ExternalPrinter {
+    sender: Sender<String>,
+    receiver: Receiver<String>,
+}
+
+#[cfg(feature = "external_printer")]
+impl ExternalPrinter {
+    /// Creates an ExternalPrinter to store lines with a max_cap
+    pub fn new(max_cap: usize) -> Self {
+        let (sender, receiver) = bounded::<String>(max_cap);
+        Self { sender, receiver }
+    }
+    /// Gets a Sender to use the printer externally by sending lines to it
+    pub fn sender(&self) -> Sender<String> {
+        self.sender.clone()
+    }
+    /// Receiver to get messages if any
+    pub fn receiver(&self) -> &Receiver<String> {
+        &self.receiver
+    }
+
+    /// Convenience method if the whole Printer is cloned, blocks if max_cap is reached.
+    pub fn print(&self, line: &str) {
+        self.sender.send(line.to_string()).unwrap_or_default();
+    }
```

Comment:
I think the general expectation is that output you want to print shouldn't get lost. 
Using a bounded channel makes certainly some sense to favor backpressure over unpredictable performance if you try to dump a ton of output.
But suppressing the `Err` case here could drop input if the receiver is done.